### PR TITLE
Jroach/add d2l navigation separator

### PR DESCRIPTION
**Pull request by jeroach**
US99013: https://rally1.rallydev.com/#/238990794704d/detail/userstory/241871087144

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,2 @@
 bower_components
-bower_components-1.x
```
[the remark this answers is not in the document]
Comment (svanherk) [reply .gitignore]: We'll want to leave these, `package-lock.json` will be created (depending on your version of node I believe) but we don't want to include it, and the `1-x` stuff stops the Polymer 1 version of the bower components from being added here as well

File: d2l-navigation-separator.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../d2l-icons/d2l-icons.html">
+
+<!--
+`d2l-navigation-separator`
+Polymer-based web component for a separator component to be used between buttons in a navigational element
+
+@demo demo/index.html
```
Comment (svanherk) [inline d2l-navigation-separator.html]: This should be `demo/d2l-navigation-separator.html`, and we'll need to update the `navigation-band.html` to point to `demo/navigation-band.html`.  This tells the docs page where to link the "Demo" button to

Comment (jeroach) [reply d2l-navigation-separator.html]: Good catch, will fix

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeroach) [reply .gitignore]: Thanks for letting me know - they have been added back.

**Comment by jeroach** [comment]
Existing `divider.scss`
http://search.dev.d2l/source/xref/GitHub/Brightspace/brightspace-integration/sass/navigation/divider.scss

Also, as per discussion with Daves, we are going to let the parent component of the separator control when it resizes (the `@media` part of the linked file. Thus, it does not need to be included here.

File: d2l-navigation-separator.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+@demo demo/d2l-navigation-separator.html
+-->
+
+<dom-module id="d2l-navigation-separator">
+	<template strip-whitespace>
+		<style>
+			span {
+				display: inline-block;
+				flex: 0 0 auto;
```
Comment (dlockhart) [inline d2l-navigation-separator.html]: The `flex` I think should stay in BSI? The fact that the LE navigation is using flex layout to position the separator is somewhat specific to it IMO.

File: d2l-navigation-separator.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+				margin: 0 9px;
+			}
+			d2l-icon {
+				color: var(--d2l-color-mica);
+			}
+		</style>
+		<span>
```
Comment (dlockhart) [inline d2l-navigation-separator.html]: Is the `span` necessary anymore? i.e. could the CSS you're applying to the `span` just go on the host element?

```
:host {
    display: inline-block;
    margin: 0 9px;
}
```

Comment (jeroach) [reply d2l-navigation-separator.html]: Does `:host` = parent? If so, then yes 😄  Just want to make sure I understand the syntax

Comment (dlockhart) [reply d2l-navigation-separator.html]: `:host` would be the `d2l-navigation-separator` element itself.

Comment (jeroach) [reply d2l-navigation-separator.html]: Ah - that should work too

Comment (jeroach) [reply d2l-navigation-separator.html]: Working on the updates to all 3 PRs

Comment (jeroach) [reply d2l-navigation-separator.html]: Done. Tested locally and looks correct 😄

File: d2l-navigation-separator.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../d2l-icons/d2l-icons.html">
```
Comment (dlockhart) [inline d2l-navigation-separator.html]: Nit: could you just import the parts of `d2l-icons` that you need? So specifically I think that would be `d2l-icon.html` and `tier2-icons.html`. It just lets the bundler optimize things a bit better by only bundling the things you actually need.

File: d2l-navigation-separator.html [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../d2l-icons/d2l-icon.html">
+<link rel="import" href="../d2l-icons/tier2-icons.html">
+
+<!--
+`d2l-navigation-separator`
+Polymer-based web component for a separator component to be used between buttons in a navigational element
+
+@demo demo/d2l-navigation-separator.html
```
Comment (svanherk) [inline d2l-navigation-separator.html]: Pretty sure this is my fault based on a previous comment.  This needs to be `@demo demo/navigation-separator.html` (without the `d2l`) to match the file name.

File: d2l-navigation-band.html [hunk 7]
```diff
@@ -5,14 +5,14 @@
 `d2l-navigation-band`
 Polymer-based web component for a solid colour band that runs along the top of the navigational header
 
-@demo demo/index.html
+@demo demo/d2l-navigation-band.html
```
Comment (svanherk) [inline d2l-navigation-band.html]: Same here, this should be `@demo demo/navigation-band.html`

Comment (jeroach) [reply d2l-navigation-band.html]: Thanks! Fixed

**Pull request merged**